src/runner/porter.runner.ts

```diff
@@ -173,6 +175,116 @@ export class PorterRunner extends Runner {
     }
   }
 
+  async portPhoneServiceInsertEvent(manager: EntityManager, batchSize = 1000): Promise<void> {
+    type LastMemberNotesType = {
+      criteria: {
+        id: FindOperator<any>;
+        appId: string;
+      };
+      lastMemberRecord: {
+        lastMemberNoteAnswered?: Date | undefined;
+        lastMemberNoteCalled?: Date | undefined;
+        lastMemberNoteCreated: Date;
+      };
+    };
+
+    type ErrInfoType = {
+      memberNote: { data: MemberNote; errMsg: string } | 'NoError';
+      member: { data: LastMemberNotesType; errMsg: string } | 'NoError';
+    };
+
+    type ErrMesType = {
+      key: string;
+      date: string;
+      info: ErrInfoType | 'No data found' | 'Data format error';
+    };
+
+    type DataType = {
+      memberNotes: MemberNote;
+      lastMemberNotes: LastMemberNotesType;
+    };
+
+    function hasDataProperty(data: DataType): boolean {
+      return Boolean(
+        data.hasOwnProperty('lastMemberNotes') &&
+          data.lastMemberNotes?.hasOwnProperty('criteria') &&
+          data.lastMemberNotes?.hasOwnProperty('lastMemberRecord') &&
+          data.hasOwnProperty('memberNotes'),
+      );
+    }
+
+    const client = this.cacheService.getClient();
+    const pattern = `PhoneService:*`;
+    const errAry: ErrMesType[] = [];
+    let cursor = '0';
+
+    do {
+      const scanResult = await client.scan(cursor, 'MATCH', pattern, 'COUNT', batchSize);
+      cursor = scanResult[0];
+      const keys = scanResult[1];
+
+      for (const key of keys) {
+        const valueString = await client.get(key);
+        const dateTime = key.split(':')[1];
+        const errMesInit: ErrMesType = {
+          key,
+          date: `${new Date(parseInt(dateTime, 10))}`,
+          info: {
+            memberNote: 'NoError',
+            member: 'NoError',
+          },
+        };
+
+        if (!valueString) {
+          errMesInit.info = 'No data found';
+          errAry.push(errMesInit);
+          continue;
+        }
+        const data: DataType = JSON.parse(valueString);
+
+        if (!hasDataProperty(data)) {
+          errMesInit.info = 'Data format error';
+          errAry.push(errMesInit);
+          continue;
+        }
+        const { memberNotes, lastMemberNotes } = data;
+        const { criteria, lastMemberRecord } = lastMemberNotes;
+
+        try {
+          await manager.getRepository(MemberNote).insert(memberNotes);
+        } catch (error) {
+          if (typeof errMesInit.info !== 'string' && errMesInit.info.memberNote === 'NoError') {
+            errMesInit.info.memberNote = { data: memberNotes, errMsg: error.toString() };
```

Comment:
It seems the current catch block's condition` if (typeof errMesInit.info !== 'string' && errMesInit.info.member === 'NoError') ` might miss catching some unexpected errors

Reply:
I have changed the code to a new implementation, avoiding misunderstanding that could lead to skipping error handling due to the 'if' statement.